lib/bambora/json_client.rb

```diff
@@ -2,24 +2,106 @@
 
 module Bambora
   class JSONClient < Bambora::Client
-    def request(method:, path: '', params: {}, body: {}, headers: {})
```

Comment:
After causing a bug by accidentally removing `to_json.to_s` on ln 10, I chose to add faraday as an interface to Excon. It wasn't until I was reviewing this PR that the error I was seeing was actually my mistake 🤦‍♀ 

Adding Faraday did force me to make explicit methods for each request verb here, which I think makes the interface to each method more clear.

Reply:
Ultimately, what inspired this change is that I needed one more arg for this method and rubocop got mad about the number of args. I considered using a hash instead, but the internals of the method got kind of messy when I did that.

Reply:
I like the way this turned out. Much less configuration on our part to get various HTTP verbs to work correctly. `faraday` gives us what we need out of the box, and still uses `excon`. I think this is a great update and will help future us.